pkg/expression/constant.go

```diff
@@ -149,9 +150,14 @@ func (c *Constant) StringWithCtx(ctx ParamValues) string {
 		}
 		c.Value.SetValue(dt.GetValue(), c.RetType)
 	} else if c.DeferredExpr != nil {
-		return c.DeferredExpr.StringWithCtx(ctx)
+		return c.DeferredExpr.StringWithCtx(ctx, redact)
 	}
-	return fmt.Sprintf("%v", c.Value.GetValue())
+	if redact == perrors.RedactLogDisable {
+		return fmt.Sprintf("%v", c.Value.GetValue())
+	} else if redact == perrors.RedactLogMarker {
+		return fmt.Sprintf("‹%v›", c.Value.GetValue())
+	}
+	return "?"
```

Comment:
String comparison is slow. Why not use a custom type like `type RedactOpt int8` instead?

Reply:
The root cause is that this enum was originally defined as a string. This also involves the slow log and other sections controlled by tidb_redact_log. Therefore, if we need to make changes, it would be better to create a new PR to address the issue